crypto/rsa/rsa_sp800_56b_gen.c

```diff
@@ -209,6 +209,143 @@ static int rsa_validate_rng_strength(EVP_RAND_CTX *rng, int nbits)
     return 1;
 }
 
+
+DEFINE_STACK_OF(BIGNUM)
+
+/*
+ *
+ * Using p & q, calculate other required parameters such as n, d.
```

Comment:
The code below looks like it assumes d is set.. So there are a lot of comments here that are no longer valid.
(I dont know whether you should recalculate the d to do a comparison or not)..
Not sure if it is feasible or not but I guess that another way of doing this might be to save some params  and just run the old call to generate the crt params, then check that the regenerated params (n & d) match if they already existed?

Reply:
I don't think d should be recalculated here, as it it is provided as an input parameter (generated in the case the keygen path is taken, explicitly provided by the user in the event the fromdata path is the source, but I've cleaned up the comments